### DISCUSSION TITLE
UGS connects but, won't home/jog on Bob's CNC KL744

### DISCUSSION THREAD

**Discussion by Hoosier-Design** [winder/Universal-G-Code-Sender#2590] [Help]
Hey, I'm a complete newbie to CNC and just finished building my Bob's CNC KL744.  I'm able to connect in UGS to the machine but, can't home or jog the machine.  I receive ALARM:9 now where it can't find the limit switch.  I've increased max travel, decreased pull-off, and checked my wiring but, can't figure it out. I moved it manually out of home when these issues started.  In the home position I'm getting all my sensor lights but no jog movements.  The home position lets me get to IDLE with alarms for XYZ but, if I click home machine I get ALARM:8.

Yesterday I was receiving ALARM:8 and ran the recommended fixes. I was able to get it to jog X (+ and -) and Y (+) but not Y (-) or Z (+ and -).  I even downloaded the software to another computer, swapped cables, and tried again.  The other computer began receiving the ALARM:9 but, also couldn't jog at all. 

I'd be super grateful for any help or advice! I'm hoping to have the machine operational for the weekend to play around with some designs with my kids.

**Comment by breiler** [comment]
The Alarm 8 means that it will trigger the limit switch while homing, but it is not able to "un-trigger" it using the configured pull off distance.

This is done using the `$27` setting which should be in millimeters. Type `$27` in the console to get the current value then increase it with for instance `$27=10`.

The Alarm 9 means that it could not find the limit switches during homing with the given search distance. The search distance is set with `$130` for X, `$131` for Y and `$132` for Z in millimeters. If for instance your machine width is 400mm - the `$130` setting needs to be at least that.

> I moved it manually out of home when these issues started.

Excellent, keep doing that until you can do a proper homing.

> I was able to get it to jog X (+ and -) and Y (+) but not Y (-) or Z (+ and -).

This could be caused by the machine not being homed. It could also mean that you have the setting `$20` enabled for soft limits. This will prevent the machine to move beyond the limits configured with `$130`, `$131` and `$131`. If you want to use homing that needs to be properly configured first and/or disable the soft limits.

The available settings are described here:
https://github.com/gnea/grbl/wiki/Grbl-v1.1-Configuration

Also take a look at the documentation for your machine:
https://www.bobscnc.com/blogs/assembly-manuals-and-documentation/kl744-documentation

**Reply by Hoosier-Design** [reply]
Hey thank you for the prompt reply! So I've, switched the settings as recommended (copied below).  I get ALARM:9 again when selecting "home machine" with no movement at all on the axes.  Jog is also still unresponsive.  I disconnected and reconnected as well as closing it and reopening with the same results (ALARM:9).  

$0 = 10    (Step pulse time, microseconds)
$1 = 255    (Step idle delay, milliseconds)
$2 = 0    (Step pulse invert, mask)
$3 = 12    (Step direction invert, mask)
$4 = 0    (Invert step enable pin, boolean)
$5 = 0    (Invert limit pins, boolean)
$6 = 0    (Invert probe pin, boolean)
$10 = 1    (Status report options, mask)
$11 = 0.010    (Junction deviation, millimeters)
$12 = 0.002    (Arc tolerance, millimeters)
$13 = 1    (Report in inches, boolean)
$20 = 0    (Soft limits enable, boolean)
$21 = 0    (Hard limits enable, boolean)
$22 = 1    (Homing cycle enable, boolean)
$23 = 3    (Homing direction invert, mask)
$24 = 250.000    (Homing locate feed rate, mm/min)
$25 = 4000.000    (Homing search seek rate, mm/min)
$26 = 250    (Homing switch debounce delay, milliseconds)
$27 = 10.000    (Homing switch pull-off distance, millimeters)
$30 = 1000    (Maximum spindle speed, RPM)
$31 = 0    (Minimum spindle speed, RPM)
$32 = 0    (Laser-mode enable, boolean)
$100 = 40.000    (X-axis travel resolution, step/mm)
$101 = 40.000    (Y-axis travel resolution, step/mm)
$102 = 500.000    (Z-axis travel resolution, step/mm)
$110 = 25400.000    (X-axis maximum rate, mm/min)
$111 = 25400.000    (Y-axis maximum rate, mm/min)
$112 = 2000.000    (Z-axis maximum rate, mm/min)
$120 = 500.000    (X-axis acceleration, mm/sec^2)
$121 = 500.000    (Y-axis acceleration, mm/sec^2)
$122 = 500.000    (Z-axis acceleration, mm/sec^2)
$130 = 1220.000    (X-axis maximum travel, millimeters)
$131 = 1220.000    (Y-axis maximum travel, millimeters)
$132 = 127.000    (Z-axis maximum travel, millimeters)

**Reply by breiler** [reply]
Do you hear the motors engaging?

If you hear the motors enaging, and you can feel/hear them vibrating when issuing homing this could mean that the speed is set to high. Lower this setting `$25 = 4000.000` to something like `1000`.


Is the emergency stop pressed - pull it out?

Depending how it is wired it may cut power to the stepper motor drivers.

**Reply by Hoosier-Design** [reply]
The motors are definitely getting power but, don't sound like they're engaging or moving.  I turned $25 down to 1000 and tried running it only to get ALARM:9 again.  I pushed in and pulled out the release just to see if it got stuck and no luck.   

>>> $25=1000
ok
>>> $H
An error was detected while sending '$H': (ALARM:9) Homing fail. Could not find limit switch within search distances. Try increasing max travel, decreasing pull-off distance, or check wiring. Streaming has been paused.
Error while processing response <ok>: An unexpected command was completed by the controller.
Grbl 1.1h BobsCNC KL733 ['$' for help]
[MSG:'$H'|'$X' to unlock]

**Reply by breiler** [reply]
No idea, I think that you should reach out to Bobs support.

**Reply by Hoosier-Design** [reply]
I appreciate your time and help.  I'll definitely send them a message.  Hopefully we can figure it out.  If we do should I post the solution here? (First time posting here)

**Reply by breiler** [reply]
Sure!
I am curious what this could be and it might help someone else with similar problems.

**Reply by Hoosier-Design** [reply]
So after a very lengthy phone call with Bob we got it resolved.  Turns out I suck at wiring.  I had a loose connection in one of the grounding wires. Once we got the wiring pulled out and reseated more securely - BOOM! - it works!.  Thank you for helping and I hope this helps someone else.

**Reply by breiler** [reply]
I am glad that you figured it out, happy machining!
